pkg/kv/kvserver/replica_raftstorage.go

```diff
@@ -421,31 +421,43 @@ func (r *Replica) updateRangeInfo(ctx context.Context, desc *roachpb.RangeDescri
 	return nil
 }
 
+type clearRangeDataOptions struct {
+	// ClearReplicatedByRangeID indicates that replicated RangeID-based keys
+	// (abort span, etc) should be removed.
+	ClearReplicatedByRangeID bool
+	// ClearUnreplicatedByRangeID indicates that unreplicated RangeID-based keys
+	// (logstore state incl. HardState, etc) should be removed.
+	ClearUnreplicatedByRangeID bool
+	// ClearReplicatedBySpan causes the state machine data (i.e. the replicated state
+	// for the given RSpan) that is key-addressable (i.e. range descriptor, user keys,
+	// locks) to be removed. No data is removed if this is the zero span.
+	ClearReplicatedBySpan roachpb.RSpan
+
+	// If MustUseClearRange is true, a Pebble range tombstone will always be used
+	// to clear the key spans (unless empty). This is typically used when we need
+	// to write additional keys to an SST after this clear, e.g. a replica
+	// tombstone, since keys must be written in order. When this is false, a
+	// heuristic will be used instead.
+	MustUseClearRange bool
```

Comment:
I think the confusion is that the method is named `clearRangeData`. The option is named adequately: when set, we use `storage.ClearRangeWithHeuristic` to actually remove the data. This is not a MVCC Range Deletion Tombstone, so I would steer clear of anything that might suggest that.
Instead of making a change now, I added a TODO to rename `clearRangeData`:

> // TODO(tbg): could rename this to clearReplicaData. The use of "Range" in both the
// "CRDB Range" and "storage.ClearRange" context in the setting of this method could
// be confusing.